extensions/logging-gelf/runtime/src/main/java/io/quarkus/logging/gelf/graal/JulLogEventSubstitution.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.logging.gelf.graal;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.LogRecord;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "biz.paluch.logging.gelf.jul.JulLogEvent")
+public final class JulLogEventSubstitution {
+    @Alias
+    private static Map<String, String> threadNameCache;
+
+    @Substitute
+    private String getThreadName(LogRecord record) {
+        // the default implementation uses the ThreadMXBean to get the name of the thread corresponding to this thead ID
+        // SVM didn't support JMX so we use Thread.getAllStackTraces().keySet() instead
+        // The following PR fixes the issue: https://github.com/mp911de/logstash-gelf/issues/213
+        // This substitution implements the same while waiting for a new release of logstash-gelf library
```

Comment:
Again, this whole thing should be replaced with the cast to `ExtLogRecord`.  Even if upstream isn't there yet.  It will be far lest costly in this case.

Reply:
done